aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -280,18 +297,83 @@ module AptosFramework::TokenV1 {
         collection: vector<u8>,
         name: vector<u8>,
         amount: u64,
+        serial_number: u64
     ) acquires TokenStore {
-        let token_id = create_token_id_raw(creators_address, collection, name);
+        let token_id = create_token_id_raw(creators_address, collection, name, serial_number);
         direct_transfer(sender, receiver, token_id, amount);
     }
 
     public(script) fun initialize_token_script(account: &signer) {
         initialize_token_store(account);
     }
 
-    //
-    // Public functions for holding tokens
-    //
+    /// mutate the token property and save the new property in TokenStore
+    /// if the token serial_number is 0, we will create a new serial number per each token and store the properties
+    /// if the token serial_number is not 0, we will just update the propertyMap
+    public fun mutate_token_properties(
```

Comment:
Does it make sense to have a single function that either mints a ton of tokens or just updates the properties of a single token? I was thinking having 2 separate functions might make more sense?

Reply:
mint is dedicated for adding more tokens for serial_number = 0.
The mutate_token_properties change the property and add a new token with a new serial_number, which is only used in mutate_token_properties.